src/main/java/com/synopsys/integration/jenkins/scan/service/bitbucket/BitBucketRepositoryService.java

```diff
@@ -0,0 +1,128 @@
+
+        if (jenkins != null) {
+
+            SCMSource scmSource = findSCMSource(jenkins, jobName);
+
+            if (scmSource instanceof BitbucketSCMSource) {
+                BitbucketSCMSource bitbucketSCMSource = (BitbucketSCMSource) scmSource;
+
+                // Creating and using them to get the bitbucket token
+                FreeStyleProject freeStyleProject = new FreeStyleProject(jenkins, jobName);
+                FreeStyleBuild freeStyleBuild = new FreeStyleBuild(freeStyleProject);
+
+                // If we are not able to get bitbucket token from the input directly then we will get it through the bitbucketSCMSource CredentialId
+                String bitBucketToken = (String) scanParameters.get("bitbucket_token");
+                if(bitBucketToken == null) {
+                    bitBucketToken = getCredentialsToken(bitbucketSCMSource.getCredentialsId(), freeStyleBuild);
+                }
+
+                BitbucketApi bitbucketApi = bitbucketSCMSource.buildBitbucketClient(bitbucketSCMSource.getRepoOwner(), bitbucketSCMSource.getRepository());
+
+                // Access the repository details with BitbucketApi
+                BitbucketRepository bitbucketRepository = bitbucketApi.getRepository();
+
+                listener.getLogger().println("Repository Name: " + bitbucketRepository.getRepositoryName());
+
+                Api bitBucketApi = new Api();
+                bitBucketApi.setUrl(bitbucketSCMSource.getServerUrl());
+                bitBucketApi.setToken(bitBucketToken);
+
+                Pull pull = new Pull();
+                pull.setNumber(projectRepositoryPullNumber);
+
+                Repository repository = new Repository();
+                repository.setName(bitbucketRepository.getRepositoryName());
+                repository.setPull(pull);
+
+                Project project = new Project();
+                project.setKey(bitbucketRepository.getProject().getKey());
+                project.setRepository(repository);
+
+                bitBucket.setApi(bitBucketApi);
+                bitBucket.setProject(project);
+
+            } else {
+                listener.getLogger().println("SCM source is not a BitbucketSCMSource.");
+            }
+        } else {
+            listener.getLogger().println("Jenkins instance not found.");
+        }
+
+        return bitBucket;
+
+    }
+
+    private static SCMSource findSCMSource(Jenkins jenkins, String jobName) {
+        SCMSourceOwner owner = jenkins.getItemByFullName(jobName, SCMSourceOwner.class);
+        if (owner != null) {
+            for (SCMSource scmSource : owner.getSCMSources()) {
+                // Check if the SCM source belongs to the job
+                if (owner.getSCMSource(scmSource.getId()) != null) {
+                    return scmSource;
+                }
+            }
+        }
+        return null;
+    }
+
+    private static String getCredentialsToken(String credentialsId, FreeStyleBuild freeStyleBuild) {
```

Comment:
we may need to move this method to somewhere common to create the space for the future changes